tests/test_declarative_assembly_format.py

```diff
@@ -0,0 +1,241 @@
+from __future__ import annotations
+
+from io import StringIO
+
+import pytest
+
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.ir.core import MLContext, Operation
+from xdsl.irdl import IRDLOperation, irdl_op_definition
+from xdsl.parser.core import Parser
+from xdsl.printer import Printer
+from xdsl.utils.exceptions import PyRDLOpDefinitionError
+
+################################################################################
+# Utils for this test file                                                     #
+################################################################################
+
+
+def check_roundtrip(program: str, ctx: MLContext):
```

Comment:
I feel like I've seen this function before. Maybe worth factoring out into a helper file?

Reply:
Oh, that comment was from a change in the other patch (since Github is really annoying with stacked PRs).
I'll keep that in mind!